File: julia/eliminacao01.jl
#Essa função de eliminação retorna U, depois de fazer a decomposição LU.
#Função faz pivotamento.


############
#Observação#
############

# Para usar essa função é necessario mandar uma matriz AA que seja do tipo float
# tentei fazer a conversão de tipo porem os resultados saiam como se fossem um número inteiro.

function eliminacao01(A)
	A
	epsilon = 0.00001;
	(m, n) = size(A);
	
	for i = 1:n-1
		pivot = A[i,i];

		if abs(pivot) < epsilon

			for k = i+1:m
				if abs(A[k,i] > epsilon)

					pivot = A[k,i];
					auxlinha = copy(A[k,:]);
					A[k,:] = copy(A[i,:]);
					A[i,:] = auxlinha;
					break

				end
			end
		end


		for j = i+1:m

			mult = A[j,i]/pivot;
			A[j,:] = A[j,:] - mult*A[i,:];
			

		end
		
	end

	return A;


end

File: python/eliminacao01.py
from numpy import *

def eliminacao01(A):

	(m , n) = A.shape
	
	for i in range(0, n):

		pivot = A[i,i]

		if pivot == 0:
			for k in range(i+1, m):
				if A[k,i] != 0:

					pivot = A[k,i]
					auxLinha = A[k].copy()
					A[k] = A[i]
					A[i] = auxLinha
					
					
					break
				else:
					continue

		for j in range(i+1, m):
			
			A[j] = A[j] - (A[j,i]/pivot)*A[i]
			
			

	return A


A = random.randint(low = 1., high = 10., size = (5,5))
#A = array( [ [0., 0., 4., 3.], [7., 1., 8., 3.], [6., 7., 3., 6.] ,[5., 2., 8., 4.] ])
#A = array( [ [2., 6., 4., 3.], [7., 1., 8., 3.], [6., 7., 3., 6.] ,[5., 2., 8., 4.] ])
#print(A.round(decimals = 2), '\n\n')
U = eliminacao01(A)



print(U.round(decimals = 2))